Also, since we need to run this PR on CI, I need you to fix all the checks first, otherwise I am not able to push this branch and trigger our CI to run on it. In this case, `CheckStyle` fails with the below:

```
> Task :WordPress:checkstyle
[ant:checkstyle] [ERROR] /Users/.../WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java:31:32: Name 'binding' must match pattern '^m[a-zA-Z0-9]*$'. [MemberName]
[ant:checkstyle] [ERROR] /Users/.../WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java:55:55: ',' is not followed by whitespace. [WhitespaceAfter]
[ant:checkstyle] [ERROR] /Users/.../WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java:55:60: ',' is not followed by whitespace. [WhitespaceAfter]
```

To run `CheckStyle` locally use this command: `./gradlew checkstyle`

PS: I also suggest that you run all checks locally before pushing and opening any PR, just to make sure our CI will also succeed afterwards (see [pipeline.yml](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/.buildkite/pipeline.yml)). For you, the most important checks and tests command we use on CI are the following:
1. `./gradlew checkstyle` for Java classes
2. `./gradlew detekt` for Kotlin classes
3. `./gradlew lintWordpressJalapenoDebug` + `./gradlew lintJetpackJalapenoDebug` for Android Lint
4. `./gradlew testWordpressJalapenoDebugUnitTest` + `./gradlew testJetpackJalapenoDebugUnitTest` for Unit Tests

I hope the above will help you... 🙏 

-----

I am going to pause reviewing #20941 until we are done with this PR first, and maybe then, after you apply what you learned here on this other PR, I'll go for another round of review on it. I hope that's okay with you. 😊

--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java
@@ -5,8 +5,6 @@ import android.content.DialogInterface;
 import android.os.Bundle;
 import android.view.LayoutInflater;
 import android.view.View;
-import android.widget.EditText;
-import android.widget.Spinner;
 
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AlertDialog;
@@ -17,6 +15,7 @@ import com.google.android.material.dialog.MaterialAlertDialogBuilder;
 
 import org.wordpress.android.R;
 import org.wordpress.android.WordPress;
+import org.wordpress.android.databinding.AddCategoryBinding;
 import org.wordpress.android.fluxc.model.SiteModel;
 import org.wordpress.android.fluxc.model.TermModel;
 import org.wordpress.android.fluxc.store.TaxonomyStore;
@@ -29,8 +28,7 @@ import javax.inject.Inject;
 
 public class AddCategoryFragment extends AppCompatDialogFragment {
     private SiteModel mSite;
-    private EditText mCategoryEditText;
-    private Spinner mParentSpinner;
+    private AddCategoryBinding mBinding;
 
     @Inject TaxonomyStore mTaxonomyStore;
 
@@ -53,16 +51,12 @@ public class AddCategoryFragment extends AppCompatDialogFragment {
                 new MaterialAlertDialogBuilder(new ContextThemeWrapper(getActivity(), R.style.PostSettingsTheme));
         // Get the layout inflater
         LayoutInflater inflater = requireActivity().getLayoutInflater();
-
-        // Inflate view
         //noinspection InflateParams
-        View view = inflater.inflate(R.layout.add_category, null);
-        mCategoryEditText = (EditText) view.findViewById(R.id.category_name);
-        mParentSpinner = (Spinner) view.findViewById(R.id.parent_category);
+        mBinding = AddCategoryBinding.inflate(inflater, null, false);
 
         loadCategories();
 
-        builder.setView(view)
+        builder.setView(mBinding.getRoot())
                .setPositiveButton(android.R.string.ok, null)
                .setNegativeButton(android.R.string.cancel, null);
 
@@ -101,12 +95,12 @@ public class AddCategoryFragment extends AppCompatDialogFragment {
     }
 
     private boolean addCategory() {
-        String categoryName = mCategoryEditText.getText().toString();
-        CategoryNode selectedCategory = (CategoryNode) mParentSpinner.getSelectedItem();
+        String categoryName = mBinding.categoryName.getText().toString();
+        CategoryNode selectedCategory = (CategoryNode) mBinding.parentCategory.getSelectedItem();
         long parentId = (selectedCategory != null) ? selectedCategory.getCategoryId() : 0;
 
         if (categoryName.replaceAll(" ", "").equals("")) {
-            mCategoryEditText.setError(getText(R.string.cat_name_required));
+            mBinding.categoryName.setError(getText(R.string.cat_name_required));
             return false;
         }
 
@@ -127,7 +121,7 @@ public class AddCategoryFragment extends AppCompatDialogFragment {
         if (categoryLevels.size() > 0) {
             ParentCategorySpinnerAdapter categoryAdapter =
                     new ParentCategorySpinnerAdapter(getActivity(), R.layout.categories_row_parent, categoryLevels);
-            mParentSpinner.setAdapter(categoryAdapter);
+            mBinding.parentCategory.setAdapter(categoryAdapter);
         }
     }
 
@@ -135,5 +129,11 @@ public class AddCategoryFragment extends AppCompatDialogFragment {
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
         outState.putSerializable(WordPress.SITE, mSite);
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        mBinding = null;
     }
 }

